manager/state/raft/transport/peer.go

```diff
@@ -132,6 +138,56 @@ func (p *peer) resolveAddr(ctx context.Context, id uint64) (string, error) {
 	return resp.Addr, nil
 }
 
+// Returns the raft message struct size (not including the payload size) for the given raftpb.Message.
+// The payload is typically the snapshot or append entries.
+func raftMessageStructSize(m *raftpb.Message) int {
+	return (&api.ProcessRaftMessageRequest{Message: m}).Size() - len(m.Snapshot.Data)
+}
+
+// Returns the max allowable payload based on MaxRaftMsgSize and
+// the struct size for the given raftpb.Message and
+func raftMessagePayloadSize(m *raftpb.Message) int {
+	return GrpcMaxMsgSize - raftMessageStructSize(m)
+}
+
+// Split a large raft message into smaller messages.
+// Currently this means splitting the []Snapshot.Data into chunks whose size
+// is dictacted by MaxRaftMsgSize.
+func splitSnapshotData(ctx context.Context, m *raftpb.Message) []api.ProcessRaftMessageRequest {
+	var messages []api.ProcessRaftMessageRequest
+	if m.Type != raftpb.MsgSnap {
+		return messages
+	}
+
+	// get the size of the data to be split.
+	size := len(m.Snapshot.Data)
+
+	// Get the max payload size.
+	payloadSize := raftMessagePayloadSize(m)
+
+	// split the snpashot into smaller messages.
+	for snapDataIndex := 0; snapDataIndex < size; {
+		remainingSize := size - snapDataIndex
+		chunkSize := int(math.Min(float64(remainingSize), float64(payloadSize)))
```

Comment:
Using floating point math here doesn't feel right.

Reply:
holy shit I know!

you'd prefer just doing if/else ? @aaronlehmann

Reply:
holy shit I know right!

you'd prefer just doing if/else ? @aaronlehmann